### ISSUE TITLE
Camera freeze when scanning QR Code

### ISSUE THREAD

**Issue by Chiheisen** [beemdevelopment/Aegis#448]
###### Info

* __Version__: 1.1.4
* __Source__: F-Droid
* __Vault encrypted:__ Yes, bio yes
* __Device:__ Asus Zenfone 6
* __Android version and ROM:__ Android 10, LineageOS 17.1

###### Steps to reproduce

1. Press + to add new entry
2. Select scan QR Code
3. Camera opens, works for a split second then freezes

###### What do you expect to happen?

No camera freeze.

###### What happens instead?

Camera moves initially for a split second, then freezes, displaying the current capture. Even if focused on a QR Code, scan doesn't complete. 
Pulling down notification slide allows it to move again for a split second, then freezes again.

###### Log

```
05-30 19:34:05.536  5148  5148 W ActivityThread: handleWindowVisibility: no activity for token android.os.BinderProxy@4759879
05-30 19:34:05.580  5148  5148 E CameraPreview: java.io.IOException: setPreviewTexture failed
05-30 19:34:05.580  5148  5148 E CameraPreview: java.io.IOException: setPreviewTexture failed
05-30 19:34:05.580  5148  5148 E CameraPreview: 	at android.hardware.Camera.setPreviewSurface(Native Method)
05-30 19:34:05.580  5148  5148 E CameraPreview: 	at android.hardware.Camera.setPreviewDisplay(Camera.java:746)
05-30 19:34:05.580  5148  5148 E CameraPreview: 	at me.dm7.barcodescanner.core.CameraPreview.showCameraPreview(CameraPreview.java:88)
05-30 19:34:05.580  5148  5148 E CameraPreview: 	at me.dm7.barcodescanner.core.CameraPreview.surfaceChanged(CameraPreview.java:73)
05-30 19:34:05.580  5148  5148 E CameraPreview: 	at android.view.SurfaceView.updateSurface(SurfaceView.java:916)
05-30 19:34:05.580  5148  5148 E CameraPreview: 	at android.view.SurfaceView.lambda$new$0$SurfaceView(SurfaceView.java:146)
05-30 19:34:05.580  5148  5148 E CameraPreview: 	at android.view.-$$Lambda$SurfaceView$w68OV7dB_zKVNsA-r0IrAUtyWas.onPreDraw(Unknown Source:2)
05-30 19:34:05.580  5148  5148 E CameraPreview: 	at android.view.ViewTreeObserver.dispatchOnPreDraw(ViewTreeObserver.java:1088)
05-30 19:34:05.580  5148  5148 E CameraPreview: 	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2769)
05-30 19:34:05.580  5148  5148 E CameraPreview: 	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1745)
05-30 19:34:05.580  5148  5148 E CameraPreview: 	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7768)
05-30 19:34:05.580  5148  5148 E CameraPreview: 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:967)
05-30 19:34:05.580  5148  5148 E CameraPreview: 	at android.view.Choreographer.doCallbacks(Choreographer.java:791)
05-30 19:34:05.580  5148  5148 E CameraPreview: 	at android.view.Choreographer.doFrame(Choreographer.java:726)
05-30 19:34:05.580  5148  5148 E CameraPreview: 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:952)
05-30 19:34:05.580  5148  5148 E CameraPreview: 	at android.os.Handler.handleCallback(Handler.java:883)
05-30 19:34:05.580  5148  5148 E CameraPreview: 	at android.os.Handler.dispatchMessage(Handler.java:100)
05-30 19:34:05.580  5148  5148 E CameraPreview: 	at android.os.Looper.loop(Looper.java:214)
05-30 19:34:05.580  5148  5148 E CameraPreview: 	at android.app.ActivityThread.main(ActivityThread.java:7356)
05-30 19:34:05.580  5148  5148 E CameraPreview: 	at java.lang.reflect.Method.invoke(Native Method)
05-30 19:34:05.580  5148  5148 E CameraPreview: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
05-30 19:34:05.580  5148  5148 E CameraPreview: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
05-30 19:34:05.589  5148  5148 E CameraPreview: java.io.IOException: setPreviewTexture failed
05-30 19:34:05.589  5148  5148 E CameraPreview: java.io.IOException: setPreviewTexture failed
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.hardware.Camera.setPreviewSurface(Native Method)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.hardware.Camera.setPreviewDisplay(Camera.java:746)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at me.dm7.barcodescanner.core.CameraPreview.showCameraPreview(CameraPreview.java:88)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at me.dm7.barcodescanner.core.CameraPreview.surfaceChanged(CameraPreview.java:73)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.view.SurfaceView.updateSurface(SurfaceView.java:916)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.view.SurfaceView.setFrame(SurfaceView.java:501)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.view.View.layout(View.java:22463)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.view.View.layout(View.java:22466)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.view.ViewGroup.layout(ViewGroup.java:6304)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.view.View.layout(View.java:22466)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.view.ViewGroup.layout(ViewGroup.java:6304)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at androidx.appcompat.widget.ActionBarOverlayLayout.onLayout(ActionBarOverlayLayout.java:446)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.view.View.layout(View.java:22466)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.view.ViewGroup.layout(ViewGroup.java:6304)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.view.View.layout(View.java:22466)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.view.ViewGroup.layout(ViewGroup.java:6304)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.view.View.layout(View.java:22466)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.view.ViewGroup.layout(ViewGroup.java:6304)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at com.android.internal.policy.DecorView.onLayout(DecorView.java:779)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.view.View.layout(View.java:22466)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.view.ViewGroup.layout(ViewGroup.java:6304)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:3157)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2614)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1745)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7768)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:967)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.view.Choreographer.doCallbacks(Choreographer.java:791)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.view.Choreographer.doFrame(Choreographer.java:726)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:952)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.os.Handler.handleCallback(Handler.java:883)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.os.Handler.dispatchMessage(Handler.java:100)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.os.Looper.loop(Looper.java:214)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at android.app.ActivityThread.main(ActivityThread.java:7356)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at java.lang.reflect.Method.invoke(Native Method)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
05-30 19:34:05.589  5148  5148 E CameraPreview: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
05-30 19:34:05.671  5148 15419 E Camera  : Error 2
05-30 19:34:05.672  5148  5148 E CameraPreview: java.lang.RuntimeException: cancelAutoFocus failed
05-30 19:34:05.672  5148  5148 E CameraPreview: java.lang.RuntimeException: cancelAutoFocus failed
05-30 19:34:05.672  5148  5148 E CameraPreview: 	at android.hardware.Camera.native_cancelAutoFocus(Native Method)
05-30 19:34:05.672  5148  5148 E CameraPreview: 	at android.hardware.Camera.cancelAutoFocus(Camera.java:1445)
05-30 19:34:05.672  5148  5148 E CameraPreview: 	at me.dm7.barcodescanner.core.CameraPreview.stopCameraPreview(CameraPreview.java:120)
05-30 19:34:05.672  5148  5148 E CameraPreview: 	at me.dm7.barcodescanner.core.CameraPreview.surfaceDestroyed(CameraPreview.java:79)
05-30 19:34:05.672  5148  5148 E CameraPreview: 	at android.view.SurfaceView.updateSurface(SurfaceView.java:861)
05-30 19:34:05.672  5148  5148 E CameraPreview: 	at android.view.SurfaceView.onWindowVisibilityChanged(SurfaceView.java:263)
05-30 19:34:05.672  5148  5148 E CameraPreview: 	at android.view.View.dispatchDetachedFromWindow(View.java:20151)
05-30 19:34:05.672  5148  5148 E CameraPreview: 	at android.view.ViewGroup.removeAllViewsInLayout(ViewGroup.java:5687)
05-30 19:34:05.672  5148  5148 E CameraPreview: 	at android.view.ViewGroup.removeAllViews(ViewGroup.java:5633)
05-30 19:34:05.672  5148  5148 E CameraPreview: 	at me.dm7.barcodescanner.core.BarcodeScannerView.setupLayout(BarcodeScannerView.java:80)
05-30 19:34:05.672  5148  5148 E CameraPreview: 	at me.dm7.barcodescanner.core.BarcodeScannerView.setupCameraPreview(BarcodeScannerView.java:187)
05-30 19:34:05.672  5148  5148 E CameraPreview: 	at me.dm7.barcodescanner.core.CameraHandlerThread$1$1.run(CameraHandlerThread.java:31)
05-30 19:34:05.672  5148  5148 E CameraPreview: 	at android.os.Handler.handleCallback(Handler.java:883)
05-30 19:34:05.672  5148  5148 E CameraPreview: 	at android.os.Handler.dispatchMessage(Handler.java:100)
05-30 19:34:05.672  5148  5148 E CameraPreview: 	at android.os.Looper.loop(Looper.java:214)
05-30 19:34:05.672  5148  5148 E CameraPreview: 	at android.app.ActivityThread.main(ActivityThread.java:7356)
05-30 19:34:05.672  5148  5148 E CameraPreview: 	at java.lang.reflect.Method.invoke(Native Method)
05-30 19:34:05.672  5148  5148 E CameraPreview: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
05-30 19:34:05.672  5148  5148 E CameraPreview: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)

```

**Comment by alexbakker** [comment]
This appears to be a crash in ``me.dm7.barcodescanner``. We have received a number of reports about strange issues with the scanner that we can't really reproduce. CameraX is supposed to have better compatibility with obscure devices, so we may want try writing our own QR scanner with CameraX and ZXing.

**Comment by alexbakker** [comment]
Hopefully fixed by #461.
